spec/takeaway_spec.rb

```diff
@@ -0,0 +1,92 @@
+require 'takeaway'
+
+describe Takeaway do
+  let(:menu) { double :menu }
+  let(:shop) { described_class.new(menu) }
+  let(:name_one) { "name one" }
+  let(:price_one) { "£0.01" }
+  let(:name_two) { "name two" }
+  let(:price_two) { "£0.02" }
+  let(:item_one) { double :item, :name => name_one, :price => price_one, :available => true }
+  let(:item_two) { double :item, :name => name_two, :price => price_two, :avaialble => true }
+
+  
+  describe '#menu' do
+    it "can be accessed" do
+      expect(shop).to respond_to(:menu)
+    end
+
+    it "displays to the customer" do
+      expect(menu).to receive(:list)
+
+      shop.menu
+    end
+  end
+
+  describe '#add_item' do
+    before do
+      expect(menu).to receive(:add)
+    end
+    it 'adds item to the menu' do
+      item_one_name = name_one
+      item_one_cost = price_one
+
+      shop.add_item(item_one_name, item_one_cost)
+    end
+  end
+
+  describe '#pick' do
+    before do
+      expect(menu).to receive(:list).and_return [item_one]
+      expect(item_one).to receive(:available?).and_return true
```

Comment:
Are you expecting or allowing? Or both at the same time? It feels like you are testing more than one behaviour here.

Reply:
Actually - you should not use expectations in before, so these are likely allows.